protostar/commands/test/environments/fuzz_test_execution_environment.py

```diff
@@ -0,0 +1,190 @@
+    TestExecutionEnvironment,
+    TestCaseCheatcodeFactory,
+)
+from protostar.commands.test.fuzzing.fuzz_input_exception_metadata import (
+    FuzzInputExceptionMetadata,
+)
+from protostar.commands.test.fuzzing.strategy_selector import StrategySelector
+from protostar.commands.test.starkware.execution_resources_summary import (
+    ExecutionResourcesSummary,
+)
+from protostar.commands.test.starkware.test_execution_state import TestExecutionState
+from protostar.commands.test.test_context import TestContextHintLocal
+from protostar.commands.test.test_environment_exceptions import ReportedException
+from protostar.commands.test.testing_seed import current_testing_seed
+from protostar.utils.data_transformer_facade import DataTransformerFacade
+
+HYPOTHESIS_VERBOSITY = Verbosity.normal
+"""
+Change this value to ``Verbosity.verbose`` while debugging Hypothesis adapter code.
+"""
+
+
+def is_fuzz_test(function_name: str, state: TestExecutionState) -> bool:
+    abi = state.contract.abi
+    params = DataTransformerFacade.get_function_parameters(abi, function_name)
+    return bool(params)
+
+
+class FuzzTestExecutionEnvironment(TestExecutionEnvironment):
+    def __init__(self, state: TestExecutionState):
+        super().__init__(state)
+
+    async def invoke(self, function_name: str) -> Optional[ExecutionResourcesSummary]:
+        abi = self.state.contract.abi
+        parameters = DataTransformerFacade.get_function_parameters(abi, function_name)
+        assert (
+            parameters
+        ), f"{self.__class__.__name__} expects at least one function parameter."
+
+        self.set_cheatcodes(
+            TestCaseCheatcodeFactory(
+                state=self.state,
+                expect_revert_context=self._expect_revert_context,
+                finish_hook=self._finish_hook,
+            )
+        )
+
+        self.set_custom_hint_locals([TestContextHintLocal(self.state.context)])
+
+        execution_resources: List[ExecutionResourcesSummary] = []
+
+        database = InMemoryExampleDatabase()
+        strategy_selector = StrategySelector(parameters)
+
+        # NOTE: Hypothesis' ``reporter`` global is a thread local variable.
+        #   Because we are running Hypothesis from separate thread, and the test itself is
+        #   running in a separate thread executor, we must set the ``reporter`` each first time
+        #   we invoke Hypothesis code in new thread.
+
+        @seed(current_testing_seed())
+        @settings(
+            database=database,
+            deadline=None,
+            print_blob=False,
+            verbosity=HYPOTHESIS_VERBOSITY,
+        )
+        @given(data_object=data())
+        async def test(data_object: DataObject):
+            with with_reporter(protostar_reporter):
+                inputs = {}
+                for param in strategy_selector.parameter_names:
+                    search_strategy = strategy_selector.search_strategies[param]
```

Comment:
We should avoid using prefixes that could be interpreted as verbs.

Reply:
SearchStrategy is a term borrowed from Hypothesis so I don't think we should rename concepts here...